network/test/block_exchange_test.go

```diff
@@ -52,7 +52,7 @@ func (suite *BlockExchangeTestSuite) SetupTest() {
 		suite.cleanupFuncs = append(suite.cleanupFuncs, cleanupFunc)
 		block := blocks.NewBlock([]byte(fmt.Sprintf("foo%v", i)))
 		suite.blockCids = append(suite.blockCids, block.Cid())
-		require.NoError(suite.T(), bstore.Put(block))
+		require.NoError(suite.T(), bstore.Put(context.Background(), block))
```

Comment:
we'd better using a `context.WithCancel` hence differing `cancel` method to make sure resources are released safely at the end of test.